src/main/java/org/broadinstitute/dsm/jobs/UPSTrackingJob.java

```diff
@@ -0,0 +1,114 @@
+        Set<String> returnTrackingIds = (HashSet<String>) ids.get("return");
+        for (String trackingId : returnTrackingIds) {
+            String transId = NanoIdUtils.randomNanoId(
+                    NanoIdUtils.DEFAULT_NUMBER_GENERATOR, "1234567890QWERTYUIOPASDFGHJKLZXCVBNM".toCharArray(), 32);
+            String inquiryNumber = trackingId;
+            String transSrc = "TestTracking";
+            String sendRequest = upsTrackingEndpoint + inquiryNumber;
+            Map<String, String> headers = new HashMap<>();
+            headers.put("transId", transId);
+            headers.put("transSrc", transSrc);
+            headers.put("Username", DSMServer.UPS_USERNAME);
+            headers.put("Password", DSMServer.UPS_PASSWORD);
+            headers.put("AccessLicenseNumber", DSMServer.UPS_ACCESSKEY);
+            headers.put("Content-Type", "application/json");
+            headers.put("Accept", "application/json");
+            try {
+                UPSTrackingResponse response = DDPRequestUtil.getResponseObjectWithCustomHeader(UPSTrackingResponse.class, sendRequest, "UPS Tracking Test " + inquiryNumber, headers);
+                logger.info("got response back: " + response);
+
+            }
+            catch (IOException e) {
+                throw new RuntimeException("couldn't get response from ups tracking ", e);
+            }
+        }
+    }
+
+
+    public static Map<String, Set<String>> getResultSet(String realm) {
+        SimpleResult result = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_SELECT_KITS)) {
+                stmt.setString(1, realm);
+                try (ResultSet rs = stmt.executeQuery()) {
+                    Map<String, Set<String>> results = getIdsFromResultSet(rs);
+                    dbVals.resultValue = results;
+                    return dbVals;
+                }
+                catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+            }
+            catch (Exception e) {
+                dbVals.resultException = e;
+            }
+            return dbVals;
+        });
+        if (result.resultException != null) {
+            throw new RuntimeException(result.resultException);
+        }
+        return (Map<String, Set<String>>) result.resultValue;
+    }
+
+    public static Map<String, Set<String>> getIdsFromResultSet(ResultSet rs) {
+        Set<String> returnTrackingIds = new HashSet<>();
+        Set<String> trackingIds = new HashSet<>();
+        try {
+            while (rs.next()) {
+                if (StringUtils.isNotBlank(rs.getString("kit.tracking_to_id"))) {
+                    trackingIds.add(rs.getString("kit.tracking_to_id"));
+                }
+                if (StringUtils.isNotBlank(rs.getString("kit.tracking_return_id"))) {
+                    returnTrackingIds.add(rs.getString("kit.tracking_return_id"));
```

Comment:
can you use the DBConstants for these strings

Reply:
Yeah sure, don't worry about UPS stuff yet, that code is not called or finished